### PR TITLE
Implement EPR exports with new DefraRubyAws gem

### PR DESCRIPTION
**Pull request by cintamani**
From https://eaflood.atlassian.net/browse/RUBY-392

In order to remove the exporter gem from the code base we have decided to also implement the EPR report as part of the 392 ticket.

This code hence includes a new rake task for epr reports in the back-office-ta projects to generate a CSV containing all currently active exemptions and upload it to an AWS bucket.

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -55,6 +55,8 @@ class Application < Rails::Application
 
     # AWS monthly bulk export
     config.bulk_reports_bucket_name = ENV["AWS_BULK_EXPORT_BUCKET"]
+    config.epr_reports_bucket_name = ENV["AWS_DAILY_EXPORT_BUCKET"]
+    config.epr_export_filename = ENV["EPR_DAYLY_REPORT_FILE_NAME"] || "waste_exemptions_epr_daily_full"
```
Comment (Cruikshanks) [inline config/application.rb]:
```suggestion
    config.epr_export_filename = ENV["EPR_DAILY_REPORT_FILE_NAME"] || "waste_exemptions_epr_daily_full"
```

File: lib/tasks/reports.rake [hunk 2]
```diff
@@ -6,5 +6,10 @@ namespace :reports do
     task bulk: :environment do
       Reports::BulkExportService.run
     end
+
+    desc "Generate the EPR reports and upload them to S3."
```
Comment (Cruikshanks) [inline lib/tasks/reports.rake]:
```suggestion
    desc "Generate the EPR report and upload it to S3."
```

File: spec/services/reports/epr_export_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module Reports
+  RSpec.describe EprExportService do
+    describe ".run" do
+      context "when the AWS request succeed" do
```
Comment (Cruikshanks) [inline spec/services/reports/epr_export_service_spec.rb]:
```suggestion
      context "when the AWS request succeeds" do
```

File: spec/services/reports/epr_export_service_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        it "generates a CSV file containing all active exemptions and upload it to AWS" do
```
Comment (Cruikshanks) [inline spec/services/reports/epr_export_service_spec.rb]:
```suggestion
        it "generates a CSV file containing all active exemptions and uploads it to AWS" do
```

File: spec/services/reports/epr_export_service_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+          # Expect no error gets notified
+          expect(Airbrake).to_not receive(:notify)
+
+          EprExportService.run
+        end
+      end
+
+      context "when the request fails" do
+        it "fails gracefully and report the error" do
```
Comment (Cruikshanks) [inline spec/services/reports/epr_export_service_spec.rb]:
```suggestion
        it "fails gracefully and reports the error" do
```

File: config/application.rb (hunk 1 above)
Comment (cintamani) [reply config/application.rb]: Ops 🙈

File: app/models/waste_exemptions_engine/registration_exemption.rb [hunk 6]
```diff
@@ -15,5 +15,15 @@ class RegistrationExemption < ActiveRecord::Base
         .includes(registration: :addresses)
         .order(:registered_on, :registration_id)
     }
+
+    scope :not_expired, -> { where("expires_on >= now()") }
```
[the remark this answers is not in the document]
Comment (cintamani) [reply app/models/waste_exemptions_engine/registration_exemption.rb]:
```suggestion
    scope :not_expired, -> { where("expires_on > CURRENT_DATE") }
```

**Pull request merged**